src/roman_datamodels/stnode.py

```diff
@@ -252,30 +261,66 @@ def get_schema(self):
             schema_uri, self.ctx, False, False)
         return schema
 
+
 class TaggedListNode(LNode):
 
     @property
     def tag(self):
         return self._tag
 
 
+_DATAMODELS_MANIFEST_PATH = importlib_resources.files(rad.resources) / "manifests" / "datamodels-1.0.yaml"
+_DATAMODELS_MANIFEST = yaml.safe_load(_DATAMODELS_MANIFEST_PATH.read_bytes())
+_OBJECT_NODE_CLASS_NAME_OVERRIDES = {
+    "tag:stsci.edu:datamodels/roman/reference_files/flat-1.0.0": "FlatRef",
+}
+_OBJECT_NODE_CLASSES = []
+
+
+def _class_name_from_tag_uri(tag_uri):
+    if tag_uri in _OBJECT_NODE_CLASS_NAME_OVERRIDES:
+        return _OBJECT_NODE_CLASS_NAME_OVERRIDES[tag_uri]
+    else:
+        tag_name = tag_uri.split("/")[-1].split("-")[0]
+        return "".join([p.capitalize() for p in tag_name.split("_")])
+
+
+for tag in _DATAMODELS_MANIFEST["tags"]:
+    class_name = _class_name_from_tag_uri(tag["tag_uri"])
+
+    docstring = ""
+    if "description" in tag:
+        docstring = tag["description"] + "\n\n"
+    docstring = docstring + f"Class generated from tag '{tag['tag_uri']}'"
+
+    cls = type(
+        class_name,
+        (TaggedObjectNode,),
+        {"_tag": tag["tag_uri"], "__module__": "roman_datamodels.stnode", "__doc__": docstring},
+    )
+    _OBJECT_NODE_CLASSES.append(cls)
+    globals()[class_name] = cls
+
+_OBJECT_NODE_CLASSES_BY_TAG = {c._tag: c for c in _OBJECT_NODE_CLASSES}
+
+
 class TaggedObjectNodeConverter(Converter):
     """
-    This class is intended to be subclassed for specific tags
+    Converter for all subclasses of TaggedObjectNode.
```

Comment:
"Converter for all subclasses of TaggedObjectNode that don't require special handling"?

Reply:
An example of a case that does use special handling (e.g., for wfi_mode, where a property for filters is added) would be good since the following uses the same converter class for all nodes. That example would be essentially done as before, and added to extensions as before?

Reply:
Oh right, we're going to need a list of exceptions or something.  I'll see what I can cook up.

Reply:
I added a custom `WfiMode` that uses properties to map `optical_element` to one of `filter` or `grating`.  This kind of implementation allows us to still use the same converter.